drivers/modem/gsm_ppp.c

```diff
@@ -76,7 +76,13 @@ static void gsm_rx(struct gsm_modem *gsm)
 	LOG_DBG("starting");
 
 	while (true) {
-		k_sem_take(&gsm->gsm_data.rx_sem, K_FOREVER);
+		/* In case semaphore gets tampered with while we're waiting for
+		 * it, periodically release and retake it.
+		 */
+		if (-EAGAIN == k_sem_take(&gsm->gsm_data.rx_sem,
```

Comment:
When we call `modem_iface_uart_init`, the `rx_sem` gets switched - it's not the same semaphore anymore. That's what I meant by "tampering". Just to make sure, I tried it with the old K_FOREVER and it doesn't work.

Reply:
Can you write the if() like this instead
```
if (k_sem_take(&gsm->gsm_data.rx_sem, K_SECONDS(1)) == -EAGAIN) {
```
as we write the checks that way in zephyr. Also comment about "When we call modem_iface_uart_init, the rx_sem gets switched so it's not the same semaphore anymore." could be useful here, otherwise future reader of this code gets to wonder what is going on here.